core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableInfo.java

```diff
@@ -97,6 +102,14 @@ public TableSchema getFactTable() {
   public void setFactTable(TableSchema factTable) {
     this.factTable = factTable;
     updateParentRelationIdentifier();
+    updateIsSchemaModified();
+  }
+
+  private void updateIsSchemaModified() {
+    if (null != factTable.getSchemaEvalution()) {
+      isSchemaModified =
```

Comment:
Add a comment here as below
If schema evolution entry list size  is > 1 that means n alter operation is performed which has added the new schema entry in the schema evolution list. Currently apart from create table schema evolution entries are getting added only in the alter operations.